python/lsst/ap/association/diaPipe.py

```diff
@@ -469,21 +506,22 @@ def createNewDiaObjects(self, diaSources):
               (`pandas.DataFrame`)
             - ``newDiaObjects`` : Newly created DiaObjects from the
               unassociated DiaSources. (`pandas.DataFrame`)
+            - ``nNewDiaObjects`` : Number of newly created diaObjects.(`int`)
         """
         newDiaObjectsList = []
-        for idx, diaSource in diaSources.iterrows():
-            if diaSource["diaObjectId"] == 0:
-                newDiaObjectsList.append(
-                    self._initialize_dia_object(diaSource["diaSourceId"]))
-                diaSources.loc[idx, "diaObjectId"] = diaSource["diaSourceId"]
+        for idx, diaSource in unAssocDiaSources.iterrows():
+            newDiaObjectsList.append(
+                self._initialize_dia_object(diaSource["diaSourceId"]))
+            unAssocDiaSources.loc[idx, "diaObjectId"] = diaSource["diaSourceId"]
```

Comment:
I'm confused here as to why you are setting the `diaObjectId` to be the `diaSourceId`.  Are these not distinct ids?

Reply:
In any case this doesn't need to be looped: you can simply say 

`unAssocDiaSources['diaObjectId'] = unAssocDiaSources['diaSourceId']`

Reply:
The id for newly created DiaObjects has been the same as the id of the first DiaSource that makes up the object since we've had DiaObjects in the pipeline. It's an easy way to give all the DiaObjects a unique id.